### PR TITLE
Fixing sections about benchmarking disk performance

### PR DESCRIPTION
**Pull request by jhutar**
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [x] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

File: guides/common/modules/proc_benchmark-disk-performance.adoc [hunk 1]
```diff
@@ -1,36 +1,41 @@
...
+* We recommend you to stop all services before executing this script and you will be prompoted to do so.
 ====
 
-This test does not use directio and will utilize the OS plus caching as normal operations would.
+This test does not use direct I/O and will utilize file caching as normal operations would.
 
 // Would it make sense to package this script for Foreman?
 You can find our first version of the script https://github.com/RedHatSatellite/satellite-support/blob/master/storage-benchmark[storage-benchmark].
-To execute just download to your {Project}, `chmod +x` the script and run:
+To execute it, just download the script to your {Project}, `chmod +x` it and run:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_benchmark-disk-performance.adoc:25]:
```suggestion
To execute it, just download the script to your {Project}, make it executable, and run:
```

File: guides/common/modules/proc_benchmark-disk-performance.adoc [hunk 2]
```diff
@@ -1,36 +1,41 @@
...
-* Local SSD based storage should values of 600MB/sec or higher.
-* Spinning disks should see values in the range of 100-200MB/sec or higher.
+* Local SSD based storage should give values of 600MB/sec or higher.
+* Spinning disks should give values in the range of 100-200MB/sec or higher.
 
 If you see values below this, please open a support ticket for assistance.
 
 ifndef::orcharhino[]
-For more information, see https://access.redhat.com/solutions/3397771[Impact of Disk Speed on {Project} Operations].
+For more information, see https://access.redhat.com/solutions/3397771[Impact of Disk Speed on Satellite Operations].
```
Comment (maximiliankolb) [inline guides/common/modules/proc_benchmark-disk-performance.adoc:40]: You cannot use downstream terminology.

`{Project}` resolves to "Satellite" for Satellite documentation: https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-satellite.adoc#L71

Alternatively, you can change from "if not orcharhino" to "if satellite" to exclude the link from other build variants.

File: guides/common/modules/ref_storage-guidelines.adoc [hunk 3]
```diff
@@ -17,8 +17,8 @@ As {ProjectName} has many operations that are I/O intensive, using high latency,
 Ensure your installation has a speed in the range 60 - 80 Megabytes per second.
 
 ifdef::satellite[]
-You can use the `fio` tool to get this data.
-See the Red{nbsp}Hat Knowledgebase solution https://access.redhat.com/solutions/3397771[Impact of Disk Speed on Satellite Operations] for more information on using the `fio` tool.
+You can use the `storage-benchmark` script to get this data.
+See the Red{nbsp}Hat Knowledgebase solution https://access.redhat.com/solutions/3397771[Impact of Disk Speed on Satellite Operations] for more information on using the `storage-benchmark` script.
```
Comment (maximiliankolb) [inline guides/common/modules/ref_storage-guidelines.adoc:21]:
```suggestion
For more information on using the `storage-benchmark` script, see https://access.redhat.com/solutions/3397771[Impact of Disk Speed on {Project} Operations].
```

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
Made some comments; the rest of your changes look good to me!

File: guides/common/modules/proc_benchmark-disk-performance.adoc (hunk 2 above)
Comment (jhutar) [reply guides/common/modules/proc_benchmark-disk-performance.adoc:40]: Hello. This is just name of the article. Article will have "Satellite" in it's name even if the documentation will be built for upstream Foreman. If you think, I'm happy to change to `if satellite`. What would be your preference?

File: guides/common/modules/ref_storage-guidelines.adoc (hunk 3 above)
Comment (jhutar) [reply guides/common/modules/ref_storage-guidelines.adoc:21]: Wording looks better, but "Impact of Disk Speed on Satellite Operations" is a name of the article. Should I add `if satellite`?

**Comment by jhutar** [comment]
Thank you for review Maximilian! Added some more comments.

File: guides/common/modules/proc_benchmark-disk-performance.adoc (hunk 2 above)
Comment (maximiliankolb) [reply guides/common/modules/proc_benchmark-disk-performance.adoc:40]: Making it "satellite only" sounds good to me!

Comment (jhutar) [reply guides/common/modules/proc_benchmark-disk-performance.adoc:40]: Ack!

**Comment by jhutar** [comment]
Hello @maximiliankolb . I think I have added pending changes now. Please take a look when you have a chance.

File: guides/common/modules/proc_benchmark-disk-performance.adoc [hunk 4]
```diff
@@ -1,36 +1,41 @@
...
 ====
-* One may have to temporarily reduce the RAM in order to run the io benchmark, aka if the box has 256 GiB that is a lot of Pulp space, so add `mem=20G` as a kernel option in grub.
-This is needed because script will execute a series of fio based IO tests against a targeted directory specified in its execution.
-This test will create a very large file that is double (2x) the size of the physical RAM on this system to ensure that we are not just testing the caching at the OS level of the storage.
-* Please bear above in mind when performing benchmark of other filesystems if you have them (like PostgreSQL storage) which might have significantly smaller capacity than Pulp storage and perhaps on different set of storage (SAN, iSCSI, etc).
+* One may have to temporarily reduce the RAM in order to run the I/O benchmark.
+For example if the host has 256 GB RAM, test would require 512 GB of storage to run.
+So as a workaround, you can add `mem=20G` kernel option in grub during system boot to temporary reduce size of system RAM.
+Test will create a very large file that is double (2x) the size of the RAM in specified directory and execute a series of storage I/O tests against it.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_benchmark-disk-performance.adoc:13]:
```suggestion
The benchmark will create a very large file that is double (2x) the size of the RAM in specified directory and execute a series of storage I/O tests against it.
```

File: guides/common/modules/proc_benchmark-disk-performance.adoc [hunk 5]
```diff
@@ -1,36 +1,41 @@
...
+The size of the file is needed to ensure that the test is not just testing the filesystem caching.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_benchmark-disk-performance.adoc:14]:
```suggestion
The size of the file ensures that the test is not just testing the filesystem caching.
```

**Review by maximiliankolb — COMMENTED** [review]
Some minor suggestions unrelated to your PR.

**Comment by jhutar** [comment]
That is perfectly relevant, thank you! Committed.

File: guides/common/modules/proc_benchmark-disk-performance.adoc [hunk 6]
```diff
@@ -1,36 +1,41 @@
...
+* One may have to temporarily reduce the RAM in order to run the I/O benchmark.
+For example if the host has 256 GB RAM, test would require 512 GB of storage to run.
+So as a workaround, you can add `mem=20G` kernel option in grub during system boot to temporary reduce size of system RAM.
+The benchmark will create a very large file that is double (2x) the size of the RAM in specified directory and execute a series of storage I/O tests against it.
+The size of the file ensures that the test is not just testing the filesystem caching.
+* Please bear above in mind when performing benchmark of other filesystems.
+For example if you plan to test on smaller volumes (typically PostgreSQL storage), you might need to reduce RAM size as described above.
+* If you are using different storage solutions (SAN, iSCSI, and so on), they might be giving different performance.
+* We recommend you to stop all services before executing this script and you will be prompoted to do so.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_benchmark-disk-performance.adoc:18]:
```suggestion
* You may have to temporarily reduce the RAM in order to run the I/O benchmark.
For example, if your {ProjectServer} has 256 GiB RAM, tests would require 512 GiB of storage to run.
As a workaround, you can add `mem=20G` kernel option in grub during system boot to temporary reduce the size of the RAM.
The benchmark creates a file twice the size of the RAM in the specified directory and executes a series of storage I/O tests against it.
The size of the file ensures that the test is not just testing the filesystem caching.
If you benchmark other filesystems, for example smaller volumes such as PostgreSQL storage, you might have to reduce the RAM size as described above.
* If you are using different storage solutions such as SAN or iSCSI, you can expect a different performance.
* {Team} recommends you to stop all services before executing this script and you will be prompoted to do so.
```

**Review by maximiliankolb — COMMENTED** [review]
Made some more suggestion.

**Comment by maximiliankolb** [comment]
Thanks. I've adjusted the commit message, merged to master, and cherry-picked: ✅ 
   15337c196..a264f5623  3.4 -> 3.4
   de7163fcf..050dd4279  3.3 -> 3.3
   0ab89363a..6ff5d957e  3.2 -> 3.2
   e75d56955..b83993d26  3.1 -> 3.1

**Pull request merged**